sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/BeamSqlSeekableTable.java

```diff
@@ -28,6 +29,7 @@
  * FROM FACT_TABLE JOIN LOOKUP_TABLE ON ...}.
  */
 public interface BeamSqlSeekableTable extends Serializable {
+  default void setJoinSubsetType(Schema joinSubsetType) {}
```

Comment:
This is error prone - while it suggests calling this method will "set JoinSubsetType", the default implementation does not do anything

I understand we cannot add "throw NotImplementedException" here either for compatibility reason.

BeamSqlSeekableTable is an interface so it does not hold members. Is there an elegant way to make joinSubsetType available in BeamSqlSeekableTable.processElement run time?

Reply:
Per my understanding, BeamSqlSeekableTable as a lookup table, it always need constructing a cache to accelerate seeking. So it's better to make joinSubsetType available before BeamSqlSeekableTable.processElement

Reply:
my concern is that having a `setJoinSubsetType` method  then `BeamSqlSeekableTable` should really be an abstract class instead of a Interface. Implying a state in interface method is antipattern (and actually Java forbids to have interface member for this reason).

pass in setUp() indeed sounds more reasonable -- in this way it will need an overwrite method for BeamSqlSeekableTable.setUp()